config/configtls/tpm_open_linux.go

```diff
@@ -0,0 +1,34 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build linux
+
+package configtls // import "go.opentelemetry.io/collector/config/configtls"
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-tpm/tpm2/transport"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+// for testing
+var tpmSimulator transport.TPMCloser
+
+func openTPM(path string) func() (transport.TPMCloser, error) {
+	return func() (transport.TPMCloser, error) {
+		if path == "" {
+			return nil, errors.New("TPM path is not set")
+		}
+		if path == "simulator" {
```

Comment:
is this standard in TPM testing?

Reply:
yes, for testing a simulator is used. To increase the coverage and keeping the TLS package clean I used this approach